.github/workflows/build-and-deploy-docs.yaml

```diff
@@ -6,7 +6,17 @@ on:
       - main
   pull_request:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
+  delete-cancelled-runs:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "MercuryTechnologies/delete-cancelled-runs@v1.0.0"
```

Comment:
Better to pin to the hash here. See https://scientific-python.org/specs/spec-0008/#pin-github-actions-release-workflows-to-their-full-release-commit-shas

Reply:
pinning any version failed, so yeah I'll try the hash anyway hahaha

Reply:
@stefanv I suppose that we don't actually need a separate job like in their docs, but that the step could be the first step of our existing build&deploy docs job?

Reply:
Yes, I think that's fine.